src/praisonai/praisonai/profiler.py

```
@@ -228,31 +237,32 @@ class Profiler:
-    _line_profile_data: Dict[str, Any] = {}
-    _cprofile_stats: List[Dict[str, Any]] = []
-    
     def __new__(cls):
         if cls._instance is None:
             with cls._lock:
                 if cls._instance is None:
                     cls._instance = super().__new__(cls)
         return cls._instance
     
+    # Class-level storage with bounded deque buffers
+    _timings = deque(maxlen=_PROFILER_MAX)
```

Ruff (0.15.12)

[warning] Mutable default value for class attribute

(RUF012)